build.gradle

```diff
@@ -172,6 +181,13 @@ tasks.named('test', Test) {
     useJUnitPlatform()
 }
 
+tasks.register('createVersionProperties') {
+    doLast {
+        File versionPropertyFile = new File(projectDir, "src/main/resources/$VERSION_PROPERTIES_FILE")
```

Comment:
I am not a gradle expert, but my understanding is that writing resource files into the `src/main/resources` directory is not the right way to include additional resources.

We already use _processResources_ to include the additional resource files from the sidebar-startpage. Maybe we could use the same, combined with a filter like discussed for example here https://stackoverflow.com/questions/40774449/gradle-processresources-with-different-replacement-and-renaming-rules? Then we could also skip the whole `clean { ... }` config part and would not have to declare another custom task. Plus no change to the .gitconfig needed too.

Reply:
From my understanding, I think we need to write the file somewhere in the end. We can create a custom dir, and use this for example. Then copy it into the jar. Using this filter, I don't understand it really.